casa/IO/test/tMultiFileLarge.cc

```diff
@@ -0,0 +1,207 @@
+    if (!useFreeBlocks  &&  itsTestMode != 0) {
+      // Reset the real size. Blocks might have been added for the header.
+      itsNrBlock = nrb1 + itsNrBlock - nrb2;
+      cout << "test mode: " << itsNrBlock << ' '<<nrb1<<' '<<nrb2<<' '<<endl;
+    }
+  }
+
+  void MultiFileLarge::writeHeaderShow (Int64 ncont, Int64 todo) const
+  {
+    if (itsTestMode != 0) {
+      int newContInx = 1 - itsHdrContInx;
+      const std::vector<Int64>& hdrContNrs = itsHdrCont[newContInx].blockNrs;
+      cout << ' ' << newContInx << itsHdrCont[0].blockNrs << itsHdrCont[1].blockNrs
+           << hdrContNrs[0] << ' ' << ncont << ' ' << todo << endl;
+    }
+  }
+
+  void MultiFileLarge::writeHeaderTest()
+  {
+    if (itsTestMode > 0  &&  itsNrBlock > itsTestMode) {
+      // Mimic an abort by not writing the first header block.
+      cout << "MultiFileLarge::writeHeader ended prematurely for nrBlock="
+           << itsNrBlock << endl;
+      exit (0);            // abort
+    }
+  }
+}
+
+
+using namespace casacore;
+
+void readFile (const String& name)
+{
+  MultiFile mfile(name, ByteIO::Old);
+  AlwaysAssertExit (! mfile.isWritable());
+  mfile.show (cout);
+  for (uInt i=0; i<mfile.info().size(); ++i) {
+    String nm = "file" + String::toString(i);
+    cout << nm << ' ' << mfile.fileId(nm, False) << endl;
+  }
+}
+
+void testLarge (Bool oDirect, Bool useCRC, int testMode)
+{
+  cout << "Test with ODirect=" << oDirect << ", useCRC=" << useCRC << endl;
+  {
+    // Use 4 KB blocks, but optionally use test mode (do not write actual data).
+    MultiFileLarge mfile("tMultiFileLarge_tmp.dat", ByteIO::New, 4096,
+                         oDirect, useCRC, testMode);
+    Int fid0 = mfile.createFile ("file0");
+    Int fid1 = mfile.createFile ("file1");
+    Vector<Int64> buf(4096/sizeof(Int64));
+    indgen(buf);
+    Int64 offs0 = 0;
+    Int64 offs1 = 0;
+    for (size_t i=0; i<1024; ++i) {
+      for (size_t j=0; j<128; ++j) {
+        mfile.write (fid0, buf.data(), 4096, offs0);
+        offs0 += 4096;
+      }
+      for (size_t j=0; j<32; ++j) {
+        mfile.write (fid1, buf.data(), 4096, offs1);
+        offs1 += 4096;
+      }
+      // Do an occasional flush.
+      if (i%200 == 5) {
+        mfile.flush();
+      }
+    }
+    AlwaysAssertExit (mfile.info()[fid0].buffer.get());
+    mfile.closeFile (fid0);
+    AlwaysAssertExit (! mfile.info()[fid0].buffer.get());
+    AlwaysAssertExit (mfile.info()[fid1].buffer.get());
+    mfile.closeFile (fid1);
+    AlwaysAssertExit (! mfile.info()[fid1].buffer.get());
+  }
+  cout << "read the large file" << endl;
+  readFile ("tMultiFileLarge_tmp.dat");
+  cout << endl;
+}
+
+
+int main (int argc, char* argv[])
+{
+  // Run as:   tMultiFileLarge testMode
+  //    testMode = 0:  do several tests for large files without writing data
+  //    testMode > 0:  create 1000000 blocks without writing data and exit prematurely
+  //    testMode < 0:  read the file produced by testMode>0
```

Comment:
I'd rather leave it as is, because it gives me the opportunity to specify the nr of blocks in the future. It's only used in the tMultiFileLarge.run file; users will hardly ever use it as such.
Besides, some translation from the command line to the enum needs to be done.